lib/elastic_apm/trace_context.rb

```diff
@@ -71,12 +64,36 @@ def hex_flags
       format('%02x', flags.to_i(2))
     end
 
+    def ensure_parent_id
+      @parent_id ||= hex_id(ID_N)
+      @parent_id
+    end
+
     def child
-      dup.tap { |tc| tc.span_id = SecureRandom.hex(8) }
+      dup.tap do |tc|
+        tc.parent_id = tc.id
+        tc.id = hex_id(ID_N)
+      end
     end
 
     def to_header
-      format('%s-%s-%s-%s', version, trace_id, span_id, hex_flags)
+      format('%s-%s-%s-%s', version, trace_id, id, hex_flags)
+    end
+
+    # @deprecated Use parent_id instead
```

Comment:
I actually have a `Deprecations` module. But that outputs logs which might not be what we want as this is more of an internally pointed deprecation?

Reply:
Internally, this is not used any more. The only reason for keeping this accessors in is for being backwards compatible in case someone used them, since `TraceContext` is part of the public API. Having this in mind, I think it would be fine if the deprecated msg would be logged. WDYT?

Reply:
I agree. Let's used the `Deprecations` module then 👍